src/plugins/saved_objects_management/public/lib/resolve_saved_objects.ts

```diff
@@ -208,9 +213,15 @@ export async function resolveIndexPatternConflicts(
       // The user decided to skip this conflict so do nothing
       return;
     }
-    obj.searchSource = await createSearchSource(indexPatterns)(
+    obj.searchSource = await createSearchSourceFactory(
       JSON.stringify(serializedSearchSource),
-      replacedReferences
+      replacedReferences,
+      indexPatterns,
+      {
+        search: getSearchService(),
+        injectedMetadata: getInjectedMetadata(),
+        uiSettings: getUiSettings(),
+      }
```

Comment:
I'm sorry having to ask that, but the service get/set pattern will have to go. All the `lib` functions in the SOM plugin has been recently adapted to remove all side effects and to use NP apis/services passed as parameters, and it will need to be the same for `resolveIndexPatternConflicts`. The services will need to be added as parameters instead and to be passed though the root view component in `src/plugins/saved_objects_management/public/management_section/mount_section.tsx` down to `src/plugins/saved_objects_management/public/management_section/objects_table/components/flyout.tsx`, where the only call to `resolveIndexPatternConflicts` is performed.